src/content/docs/de/guides/upgrade-to/v3.mdx

```diff
@@ -0,0 +1,679 @@
+
+ ```ts title="src/content/config.ts" del={1} ins={2} "({ image })"
+import { defineCollection, z, image } from "astro:content";
+import { defineCollection, z } from "astro:content";
+
+ defineCollection({
+   schema: ({ image }) =>
+     z.object({
+       image: image(),
+    }),
+});
+```
+
+### Entfernt: Shiki-Theme-Namen vor Version 0.14
+``
+
+In Astro v2.x wurden einige Shiki-Theme-Namen umbenannt, aber die ursprünglichen Namen wurden wegen der Abwärtskompatibilität beibehalten.
+
+Astro v3.0 entfernt die ursprünglichen Namen zugunsten der umbenannten Theme-Namen.
+
+#### Was soll ich tun?
+
+Wenn dein Projekt eines der folgenden Themes verwendet, benenne es entsprechend seinem aktualisierten Namen um:
+
+- `material-darker` -> `material-theme-darker`
+- `material-default` -> `material-theme`
+- `material-lighter` -> `material-theme-lighter`
+- `material-ocean` -> `material-theme-ocean`
+- `material-palenight` -> `material-theme-palenight`
+
+### Entfernt: `class:list`-Features
+
+In Astro v2.x verwendete die [`class:list`-Direktive](/de/reference/directives-reference/#classlist) eine benutzerdefinierte Implementierung, die von [`clsx`](https://github.com/lukeed/clsx) inspiriert war und einige zusätzliche Features wie Deduplizierung und Unterstützung für `Set` enthielt.
+
+Astro v3.0 verwendet `clsx` jetzt direkt für `class:list`, was keine Deduplizierung oder Unterstützung für `Set`-Werte bietet.
+
+#### Was soll ich tun?
+
+Ersetze alle `Set`-Elemente, die der `class:list`-Direktive übergeben werden, durch ein normales `Array`.
+
+```astro title="src/components/MyAstroComponent.astro" del={4} ins={5}
+<Component class:list={[
+  'a',
+  'b',
+  new Set(['c', 'd'])
+  ['c', 'd']
+]} />
+```
+
+### Entfernt: `class:list` als Prop Übergeben
+
+In Astro v2.x wurden [`class:list`-Werte](/de/reference/directives-reference/#classlist) über [`Astro.props['class:list']`](/de/reference/api-reference/#astroprops) an Komponenten übermittelt.
+
+Astro v3.0 normalisiert `class:list`-Werte zuerst zu einem String, bevor sie über `Astro.props['class']` an Komponenten weitergegeben werden.
+
+#### Was soll ich tun?
+
+Entferne jeglichen Code, der den `class:list`-Prop erhält.
+
+```astro title="src/components/MyAstroComponent.astro" del={2,3,7} ins={4,8} "classList" "'class:list': classList"
+---
+import { clsx } from 'clsx';
+const { class: className, 'class:list': classList } = Astro.props;
+const { class: className } = Astro.props;
+---
+<div
+  class:list={[className, classList]}
+  class:list={[className]}
+/>
+```
+
+### Entfernt: kebab-case Umwandlung für camelCase CSS-Variablen 
+
+In Astro v2.x wurden camelCase [CSS-Variablen](/de/guides/styling/#css-variables), die dem `style`-Attribut übergeben wurden, sowohl in camelCase (wie geschrieben) als auch im kebab-case (für Abwärtskompatibilität beibehalten) gerendert.
```

Comment:
```suggestion
In Astro v2.x wurden camelCase [CSS-Variablen](/de/guides/styling/#css-variablen), die dem `style`-Attribut übergeben wurden, sowohl in camelCase (wie geschrieben) als auch im kebab-case (für Abwärtskompatibilität beibehalten) gerendert.
```
Same issue! If the page you are linking to has translated headings, then use the German ones, not the English!  🙌